lib/http/response/body.rb

```diff
@@ -53,7 +53,10 @@ def to_s
           @streaming  = false
           @contents   = String.new("").force_encoding(encoding)
 
-          while (chunk = @stream.readpartial)
+          length = @length || Float::INFINITY
```

Comment:
@tarcieri the purpose of this variable got lost in the rebase. I was passing the content length (when available) to the body to prevent mismatches and eventual infinite loops. But the initialize signature changed. I leave the decision to you to take advantage of this or remove it.